File: yt/yt/flow/examples/python/noop/reader.py
"""Reader: RowFunction of the source computation."""

from yt.yt.flow.library.python.companion.computation import RowFunction


class Reader(RowFunction):
    """Receives the records of the source and emits nothing; put your logic here."""

    def on_message(self, message, output, ctx):
        pass

File: yt/yt/flow/examples/go/noop/reader.go
package main

import (
	"context"

	"go.ytsaurus.tech/yt/go/flow"
)

// reader receives the records of the source and emits nothing; put your logic here.
type reader struct{}

var _ flow.RowFunction = (*reader)(nil)

func (*reader) OnMessage(
	ctx context.Context,
	rt flow.Runtime,
	msg flow.ExtendedMessage,
	out flow.OutputCollector,
) error {
	return nil
}
